Sources/System/FileOperations.swift

```diff
@@ -370,4 +370,52 @@ extension FileDescriptor {
   public func dup2() throws -> FileDescriptor {
     fatalError("Not implemented")
   }
+  
+  /// A pair of `FileDescriptor` values represening a pipe.
+  ///
+  /// You are responsible for managing the lifetime and validity
+  /// of the `read` and `write` `FileDescriptor` values,
+  /// which can be closed independently or via helper methods on `Pipe`, but not both.
+  public struct Pipe {
+    /// The file descriptor for reading from this pipe
+    public let fileDescriptorForReading: FileDescriptor
+    
+    /// The file descriptor for writing to this pipe
+    public let fileDescriptorForWriting: FileDescriptor
+    
+    /// Create a pipe, a uniderctional data channel which can be used for interprocess communication.
+    ///
+    /// - Returns: A `Pipe` value representing the read and write ends of the created `Pipe`.
+    ///
+    /// The corresponding C function is `pipe`.
+    @_alwaysEmitIntoClient
+    // @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+    public init(
+      retryOnInterrupt: Bool = true
+    ) throws {
+      self = try Self._create(retryOnInterrupt: retryOnInterrupt).get()
+    }
+    
+    // @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+    @usableFromInline
+    internal static func _create(
+      retryOnInterrupt: Bool
+    ) -> Result<Pipe, Errno> {
+      var fds: [Int32] = [-1, -1]
+      return valueOrErrno(retryOnInterrupt: retryOnInterrupt) {
+        system_pipe(&fds)
+      }.map { _ in Pipe(fds: fds) }
+    }
+    
+    private init(fds: [Int32]) {
```

Comment:
I know the stdlib proper has conventions along those lines, much of it due to historical limitations. If the init doesn't appear in the ABI at all, then I'm not sure what value that convention brings for Swift System.

Reply:
I haven't contributed much to either, but I've changed this to `_input:output:` to match @lorentey's recommendations below. Let me know if I should change it to `input:output:`.

Reply:
> I know the stdlib proper has conventions along those lines, much of it due to historical limitations. If the init doesn't appear in the ABI at all, then I'm not sure what value that convention brings for Swift System.

The primary benefit of underscoring non-public names in my view is consistency and simplicity. I much prefer rules that are easy to explain, especially to folks who aren't necessarily well-versed in the intricacies of Swift's ABI features. I think it's reasonable to expect that people know what a public declaration is -- I don't think it's reasonable to expect that the average Swift developer would know exactly what declarations have an ABI impact. (Simply because it's so unusual to work on an ABI stable library.) So I think it's better to simply say that the policy is that the name of anything that's not declared `public` must carry an underscore. (The need to add an underscore also helps eliminate never-ending naming discussions for these internals, just because it makes these names "imperfect", by design, lowering stakes.)

Underscoring everything that's not public also helps prevent mistakes when previously private/internal declarations get updated to `@usableFromInline` or `public`. However, this is only a minor benefit.